Mathlib/Analysis/Meromorphic/Divisor/Basic.lean

```diff
@@ -3,231 +3,299 @@ Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+instance [AddCommGroup Y] : Sub (locallyFinsuppWithin U Y) where
   sub D₁ D₂ := mk_of_mem (D₁ - D₂) <| sub_mem D₁.memAddSubgroup D₂.memAddSubgroup
 
-instance : SMul ℕ (DivisorOn U) where
+instance [AddCommGroup Y] : SMul ℕ (locallyFinsuppWithin U Y) where
   smul n D := mk_of_mem (n • D) <| nsmul_mem D.memAddSubgroup n
 
-instance : SMul ℤ (DivisorOn U) where
+instance [AddCommGroup Y] : SMul ℤ (locallyFinsuppWithin U Y) where
   smul n D := mk_of_mem (n • D) <| zsmul_mem D.memAddSubgroup n
 
-@[simp] lemma coe_zero : ((0 : DivisorOn U) : 𝕜 → ℤ) = 0 := rfl
-@[simp] lemma coe_add (D₁ D₂ : DivisorOn U) : (↑(D₁ + D₂) : 𝕜 → ℤ) = D₁ + D₂ := rfl
-@[simp] lemma coe_neg (D : DivisorOn U) : (↑(-D) : 𝕜 → ℤ) = -(D : 𝕜 → ℤ) := rfl
-@[simp] lemma coe_sub (D₁ D₂ : DivisorOn U) : (↑(D₁ - D₂) : 𝕜 → ℤ) = D₁ - D₂ := rfl
-@[simp] lemma coe_nsmul (D : DivisorOn U) (n : ℕ) : (↑(n • D) : 𝕜 → ℤ) = n • (D : 𝕜 → ℤ) := rfl
-@[simp] lemma coe_zsmul (D : DivisorOn U) (n : ℤ) : (↑(n • D) : 𝕜 → ℤ) = n • (D : 𝕜 → ℤ) := rfl
-
-instance : AddCommGroup (DivisorOn U) :=
-  Function.Injective.addCommGroup (M₁ := DivisorOn U) (M₂ := 𝕜 → ℤ)
+@[simp] lemma coe_zero [AddCommGroup Y] :
+    ((0 : locallyFinsuppWithin U Y) : X → Y) = 0 := rfl
+@[simp] lemma coe_add [AddCommGroup Y] (D₁ D₂ : locallyFinsuppWithin U Y) :
+    (↑(D₁ + D₂) : X → Y) = D₁ + D₂ := rfl
+@[simp] lemma coe_neg [AddCommGroup Y] (D : locallyFinsuppWithin U Y) :
+    (↑(-D) : X → Y) = -(D : X → Y) := rfl
+@[simp] lemma coe_sub [AddCommGroup Y] (D₁ D₂ : locallyFinsuppWithin U Y) :
+    (↑(D₁ - D₂) : X → Y) = D₁ - D₂ := rfl
+@[simp] lemma coe_nsmul [AddCommGroup Y] (D : locallyFinsuppWithin U Y) (n : ℕ) :
+    (↑(n • D) : X → Y) = n • (D : X → Y) := rfl
+@[simp] lemma coe_zsmul [AddCommGroup Y] (D : locallyFinsuppWithin U Y) (n : ℤ) :
+    (↑(n • D) : X → Y) = n • (D : X → Y) := rfl
+
+instance [AddCommGroup Y] : AddCommGroup (locallyFinsuppWithin U Y) :=
+  Injective.addCommGroup (M₁ := locallyFinsuppWithin U Y) (M₂ := X → Y)
     _ coe_injective coe_zero coe_add coe_neg coe_sub coe_nsmul coe_zsmul
 
-instance : LE (DivisorOn U) where
-  le := fun D₁ D₂ ↦ (D₁ : 𝕜 → ℤ) ≤ D₂
+instance [LE Y] [Zero Y] : LE (locallyFinsuppWithin U Y) where
+  le := fun D₁ D₂ ↦ (D₁ : X → Y) ≤ D₂
 
-lemma le_def {D₁ D₂ : DivisorOn U} : D₁ ≤ D₂ ↔ (D₁ : 𝕜 → ℤ) ≤ (D₂ : 𝕜 → ℤ) := ⟨(·),(·)⟩
+lemma le_def [LE Y] [Zero Y] {D₁ D₂ : locallyFinsuppWithin U Y} :
+    D₁ ≤ D₂ ↔ (D₁ : X → Y) ≤ (D₂ : X → Y) := ⟨(·),(·)⟩
 
-instance : LT (DivisorOn U) where
-  lt := fun D₁ D₂ ↦ (D₁ : 𝕜 → ℤ) < D₂
+instance [Preorder Y] [Zero Y] : LT (locallyFinsuppWithin U Y) where
+  lt := fun D₁ D₂ ↦ (D₁ : X → Y) < D₂
 
-lemma lt_def {D₁ D₂ : DivisorOn U} : D₁ < D₂ ↔ (D₁ : 𝕜 → ℤ) < (D₂ : 𝕜 → ℤ) := ⟨(·),(·)⟩
+lemma lt_def [Preorder Y] [Zero Y] {D₁ D₂ : locallyFinsuppWithin U Y} :
+    D₁ < D₂ ↔ (D₁ : X → Y) < (D₂ : X → Y) := ⟨(·),(·)⟩
 
-instance : Max (DivisorOn U) where
+instance [Lattice Y] [Zero Y] : Max (locallyFinsuppWithin U Y) where
```

Comment:
Do you need the full power of `Lattice` here? Or is a [SemilatticeSup](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Lattice.html#SemilatticeSup) good enough?